### PR TITLE
feat: goal site auth api (initial)

### PR DESCRIPTION
**Pull request by ivinayakg**
# RFC - #1432 

## Brief
- We needed a simple API, that would call the goal site API with user_id, roles, GOAL_BACKEND_SECRET_KEY, and then return the token which can be used to access the goal site backend.

### Integration tests
image

### Unit tests
image

### CODE REVIEW

File: controllers/goals.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const { roles, id: userId } = req.userData;
+    const goalSiteConfig = config.services.goalAPI;
+
+    const goalApiResponse = await fetch(`${goalSiteConfig.baseUrl}/auth_token`, {
+      method: "POST",
+      body: JSON.stringify({
+        user_id: userId,
+        roles,
+        goal_api_secret_key: goalSiteConfig.secretKey,
+      }),
```
Comment (pallabez) [inline controllers/goals.js:15]: Why are we using stringify here?

Comment (ivinayakg) [reply controllers/goals.js:15]: Using the browser fetch api, there we have to send json, right?

File: controllers/goals.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            roles: roles,
+          },
+        },
+      }),
+      headers: { "Content-Type": "application/vnd.api+json", "Rest-Key": goalSiteConfig.secretKey },
+    });
+
+    const goalApiData = await goalApiResponse.json();
+
+    if (goalApiData) {
```
Comment (iamitprakash) [inline controllers/goals.js:24]: what about below code

if (goalApiResponse.status === 200) {
  const goalApiData = await goalApiResponse.json();
  return res.status(200).json({ message: "success", user: { ...data.attributes, id: data.id } });
} else {
  return res.status(goalApiResponse.status).json({ message: "error" });
}

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: controllers/goals.js (hunk 2 above)
Comment (ivinayakg) [reply controllers/goals.js:24]: sure

**Review by pallabez — APPROVED** [review]
LGTM

**Comment by ivinayakg** [comment]
> LGTM. The workflow test is failing tho.

just checked.
they are running on the local, not sure why failing here

**Comment by shubhamsigdar1** [comment]
Test is failing @ivinayakg

**Pull request merged**